wagtail/core/models.py

```diff
@@ -373,6 +373,26 @@ def get_active(cls):
         except (cls.DoesNotExist, LookupError):
             return cls.get_default()
 
+    @transaction.atomic
+    def delete(self, *args, **kwargs):
```

Comment:
Any reason to not use a `pre_delete` here? (thinking about whether this would work if someone mass-deletes using a ``Locale.objects.filter(...).delete()``?)

Reply:
I guess you missed my rant on Slack :-)

> TIL that Django `pre_delete` signals are fired after `on_delete=PROTECT` checks have been done, so you can't use them for fixing up those relations in advance of the deletion happening. _shakes fist at Django_

(a signal would have definitely been my preference, both for handling queryset deletes and keeping page-specific logic out of the Locale model...)